File: ProjetPI/src/ProjectBundle/Entity/Demande.php
<?php

namespace ProjectBundle\Entity;
use Symfony\Component\Validator\Constraints as Assert;
use Doctrine\ORM\Mapping as ORM;

/**
 * Demande
 *
 * @ORM\Table(name="demande")
 * @ORM\Entity(repositoryClass="ProjectBundle\Repository\DemandeRepository")
 */
class Demande
{
    /**
     * @var int
     *
     * @ORM\Column(name="id", type="integer")
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="AUTO")
     */
    private $id;

    /**
     * @ORM\Column(name="photo_demande", type="string", length=500)
     * @Assert\NotBlank(message="veuillez inserer une photo")
     * @Assert\File(maxSize="500k", mimeTypes={"image/jpeg", "image/jpg", "image/png", "image/GIF"}, mimeTypesMessage="Veuillez inserer une image de type jpeg, jpg, png ou GIF")
     */
    private $photoDemande;



    /**
     * @ORM\OneToMany(targetEntity="Donation", mappedBy="demandeDonation",cascade={"persist","remove"} )
     */
    private $userDemande;


    /**
     * @var enum
     *
     * @ORM\Column(name="type_demande", type="string", columnDefinition="enum('fourniture scolaire', 'vetements', 'materiel', 'nourriture')")
     * @Assert\NotBlank(message="Le type est obligatoire")
     */
    private $typeDemande;





    /**
     * @var int
     *
     * @ORM\Column(name="etat_demande", type="integer")
     */
    private $etatDemande;

    /**
     * @var string
     *
     * @ORM\Column(name="description_demande", type="string", length=255)
     * @Assert\NotBlank(message="La description est obligatoire")
     * @Assert\Length(
     *      min = 5,
     *      max = 255,
     *      minMessage = "La description doit comporter au moins 5 caractères",
     *      maxMessage = "La description doit comporter au plus 255 caractères"
     * )
     */
    private $descriptionDemande;

    /**
     * @var string
     *
     * @ORM\Column(name="titre_demande", type="string", length=255)
     * @Assert\NotBlank(message="Le titre est obligatoire")
     * @Assert\Length(
     *      min = 2,
     *      max = 20,
     *      minMessage = "Le titre doit comporter au moins 2 caractères",
     *      maxMessage = "Le titre doit comporter au plus 20 caractères"
     * )
     */
    private $titreDemande;

    /**
     * Demande constructor.
     * @param int $etatDemande
     */
    public function __construct($etatDemande)
    {
        $this->etatDemande = $etatDemande;
    }


    /**
     * Get id
     *
     * @return int
     */
    public function getId()
    {
        return $this->id;
    }

    /**
     * Set etatDemande
     *
     * @param integer $etatDemande
     *
     * @return Demande
     */
    public function setEtatDemande($etatDemande)
    {
        $this->etatDemande = $etatDemande;

        return $this;
    }

    /**
     * Get etatDemande
     *
     * @return int
     */
    public function getEtatDemande()
    {
        return $this->etatDemande;
    }

    /**
     * Set descriptionDemande
     *
     * @param string $descriptionDemande
     *
     * @return Demande
     */
    public function setDescriptionDemande($descriptionDemande)
    {
        $this->descriptionDemande = $descriptionDemande;

        return $this;
    }

    /**
     * Get descriptionDemande
     *
     * @return string
     */
    public function getDescriptionDemande()
    {
        return $this->descriptionDemande;
    }

    /**
     * Set titreDemande
     *
     * @param string $titreDemande
     *
     * @return Demande
     */
    public function setTitreDemande($titreDemande)
    {
        $this->titreDemande = $titreDemande;

        return $this;
    }

    /**
     * Get titreDemande
     *
     * @return string
     */
    public function getTitreDemande()
    {
        return $this->titreDemande;
    }

    /**
     * @return mixed
     */
    public function getUserDemande()
    {
        return $this->userDemande;
    }

    /**
     * @param mixed $user_demande
     */
    public function setUserDemande($userDemande)
    {
        $this->userDemande = $userDemande;
    }

    /**
     * @return enum
     */
    public function getTypeDemande()
    {
        return $this->typeDemande;
    }

    /**
     * @param enum $typeDemande
     */
    public function setTypeDemande($typeDemande)
    {
        $this->typeDemande = $typeDemande;
    }

    /**
     * @return mixed
     */
    public function getPhotoDemande()
    {
        return $this->photoDemande;
    }

    /**
     * @param mixed $photoDemande
     */
    public function setPhotoDemande($photoDemande)
    {
        $this->photoDemande = $photoDemande;
    }



    public function __toString(){
        // to show the name of the Category in the select
        return strval($this->id);
        // to show the id of the Category in the select
        // return $this->id;
    }


}

File: DonationMobile/src/com/mycompany/myapp/entities/Demande.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.mycompany.myapp.entities;

/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */



public class Demande {


    private Integer id;

    private String typeDemande;

    private int etatDemande;

    private String descriptionDemande;

    private String titreDemande;

    private String photoDemande;



    public Demande() {
    }

    public Demande(Integer id) {
        this.id = id;
    }

    public Demande(Integer id, int etatDemande,String typeDemande, String descriptionDemande, String titreDemande, String photoDemande) {
        this.id = id;
        this.etatDemande = etatDemande;
        this.typeDemande = typeDemande;
        this.descriptionDemande = descriptionDemande;
        this.titreDemande = titreDemande;
        this.photoDemande = photoDemande;
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getTypeDemande() {
        return typeDemande;
    }

    public void setTypeDemande(String typeDemande) {
        this.typeDemande = typeDemande;
    }

    public int getEtatDemande() {
        return etatDemande;
    }

    public void setEtatDemande(int etatDemande) {
        this.etatDemande = etatDemande;
    }

    public String getDescriptionDemande() {
        return descriptionDemande;
    }

    public void setDescriptionDemande(String descriptionDemande) {
        this.descriptionDemande = descriptionDemande;
    }

    public String getTitreDemande() {
        return titreDemande;
    }

    public void setTitreDemande(String titreDemande) {
        this.titreDemande = titreDemande;
    }

    public String getPhotoDemande() {
        return photoDemande;
    }

    public void setPhotoDemande(String photoDemande) {
        this.photoDemande = photoDemande;
    }



    @Override
    public int hashCode() {
        int hash = 0;
        hash += (id != null ? id.hashCode() : 0);
        return hash;
    }

    @Override
    public boolean equals(Object object) {
        // TODO: Warning - this method won't work in the case the id fields are not set
        if (!(object instanceof Demande)) {
            return false;
        }
        Demande other = (Demande) object;
        if ((this.id == null && other.id != null) || (this.id != null && !this.id.equals(other.id))) {
            return false;
        }
        return true;
    }

    @Override
    public String toString() {
        return "Entities.Demande[ id=" + id + titreDemande + "....." + typeDemande +" ]";
    }
    
}
